test/views/lookup/PsaLookupProtectionNotificationNoFormViewSpec.scala

```diff
@@ -0,0 +1,135 @@
+ */
+
+package views.lookup
+
+import forms.PSALookupProtectionNotificationNoForm
+import org.jsoup.Jsoup
+import play.api.i18n.Messages.Implicits._
+import testHelpers.ViewSpecHelpers.CommonViewSpecHelper
+import testHelpers.ViewSpecHelpers.lookup.PsaLookupProtectionNotificationNoFormSpecMessages
+import views.html.pages.lookup.psa_lookup_protection_notification_no_form
+
+
+class PsaLookupProtectionNotificationNoFormViewSpec extends CommonViewSpecHelper with PsaLookupProtectionNotificationNoFormSpecMessages {
+
+  "The Psa Lookup Protection Notification No Form view" when {
+
+    "provided with a form without errors" should {
+      lazy val form = PSALookupProtectionNotificationNoForm.pnnForm.bind(Map("lifetimeAllowanceReference" -> "IP141000000000A"))
+      lazy val view = psa_lookup_protection_notification_no_form(form)
+      lazy val doc = Jsoup.parse(view.body)
+
+      "have the correct title" in {
+        doc.title() shouldBe titleText
+      }
+
+      "have the correct header" in {
+        doc.select("h1").text() shouldBe headingText
+      }
+
+      "not have an error summary" in {
+        doc.select("div.error-summary").size() shouldBe 0
+      }
+
+      "have a form" which {
+
+        "has the correct method" in {
+          doc.select("form").attr("method") shouldBe "POST"
+        }
+
+        "has the correct action" in {
+          doc.select("form").attr("action") shouldBe controllers.routes.LookupController.submitProtectionNotificationNoForm().url
+        }
+      }
+
+      "have a fieldset with a div without error classes" in {
+        doc.select("fieldset div").first().attr("class") shouldBe "form-field"
+      }
+
+      "not have error notifications" in {
+        doc.select("fieldset div.error-notification").size() shouldBe 0
+      }
+
+      "have a hint with the correct text" in {
+        doc.select("p.form-field--hint").text() shouldBe hintText
+      }
+
+      "have a text input with the data pre-populated" in {
+        doc.select("input[type=text]").attr("value") shouldBe "IP141000000000A"
+      }
+
+      "have a button" which {
+
+        "has the correct text" in {
+          doc.select("button").text() shouldBe buttonText
+        }
+
+        "has the correct type" in {
+          doc.select("button").attr("type") shouldBe "submit"
+        }
+      }
+
+      "have a back-link" which {
+
+        "has the correct text" in {
+          doc.select("a.back-link").text() shouldBe plaBaseBack
+        }
+
+        "has the correct destination" in {
+          doc.select("a.back-link").attr("href") shouldBe controllers.routes.LookupController.displaySchemeAdministratorReferenceForm().url
+        }
+      }
+    }
+
+    "provided with a form with errors" should {
+      lazy val form = PSALookupProtectionNotificationNoForm.pnnForm.bind(Map("lifetimeAllowanceReference" -> "A"))
+      lazy val view = psa_lookup_protection_notification_no_form(form)
+      lazy val doc = Jsoup.parse(view.body)
+
```

Comment:
Would it be worth it to include a test to make sure the right error message was displayed?

Reply:
This would apply to the other error checks on other pages.

Reply:
The error message is not controlled by the view, it's by the form so it's not the responsibility of the view tests to cover it.

Reply:
Perhaps not the unique messages but the form always returns the a h2 with the message "Something you've entered isn't valid" and then the form error below. Shouldn't that h2 be tested since that is defined by the view?